src/api/route-services/samples.js

```diff
@@ -81,27 +82,39 @@ class SamplesService {
   async updateSamples(projectUuid, experimentId, body) {
     logger.log(`Updating samples for project ${projectUuid} and expId ${experimentId}`);
 
+    const dynamodb = createDynamoDbInstance();
+
     const marshalledKey = convertToDynamoDbRecord({
       experimentId,
     });
 
+    const getSamplesParams = {
+      TableName: this.samplesTableName,
+      Key: marshalledKey,
+      ProjectionExpression: 'samples',
+    };
+
+    // Merge samples record with the body
+    const response = await dynamodb.getItem(getSamplesParams).promise();
+    const { samples } = convertToJsObject(response.Item);
```

Comment:
Do you think it would be a lot of work to, instead of getting, merging and setting it back in dynamo, send one request that replaces the samples that were updated? 

I think that would require some changes to the endpoints too though, so perhaps the motivation for doing it this way was to just avoid the racing conditions but not perform a full fix so that we don't spend too much time on the old endpoints we are going to stop using soon? Or was there another reason?

Reply:
Unless there is an easy way to merge and update the sample cells with a DynamoDB UpdateExpression, I still think we'd need to fetch the samples, merge and insert the value back into the table.

Reply:
I agree. I'd say it's an acceptable solution for now.